rules/hmap/hmapbuild.c

```diff
@@ -55,16 +55,16 @@ int main(int ac, char **av) {
     }
     debug("Writing hmap with %u entries", numEntries);
     HeaderMap *hmap = hmap_new(numEntries);
+    int rc = 0;
     for (m = entries; m != NULL; m = m->hh.next) {
         if (hmap_addEntry(hmap, m->key, m->value)) {
             fprintf(stderr, "failed to add '%s' to hmap\n", m->key);
+            rc |= 1;
```

Comment:
AFAIK this will cause an issue for few rules I've seen internally and I think it's better to address these logic errors outside of the broken write code

Reply:
The only way `hmap_addEntry()` can return non-zero is if the table is full:

https://github.com/bazel-ios/rules_ios/blob/985d578d8d4c3e4b6fe23dc69436ecafa33d30b4/rules/hmap/hmap.c#L150-L154

I think this would be important to catch and abort no?

Reply:
Yeah it's probably fine to do would just rather batch to a smaller PR and I can resubmit this, with this impending disk cache problem I wanted to keep the changes minimal 

PS: I dug through logs and some funky build files I've seen rely on is this one:
```
static void add_entry(mapping **hashmap, char *key, char *value) {
    mapping *entry = NULL;
    HASH_FIND_STR(*hashmap, key, entry);
    if (entry) {
        // key already in the hash, verify the value
        if (strcmp(value, entry->value)) {
            fprintf(stderr,
                    "WARNING: header '%s' already in cache as '%s' ignoring "
                    "new value: '%s'\n",
                    key, entry->value, value);
        }
        return;
    }
```